pages/dashboard/index.js

```diff
@@ -93,6 +95,21 @@ export default function Dashboard() {
         };
     }, [matrixClient, matrix, invitations, setInvitations]);
 
+    const handleRemoveBookmark = async (roomId) => {
```

Comment:
Given this function is -only- being used within the `<Bookmark>` component, it should live there instead.

I suppose that the Immer-managed `matrix.bookmarks` variable will be updated by Immer.

Reply:
When doing this *and* removing `<DisplayBookmarks>` we're going to save up on two pass-downs of this function.

Reply:
but wouldn't we instead create a new functions with every render of `DisplayBookmarks`? so for every single bookmark? (same with `getOrigin()`

Reply:
I know what you mean, from the "memory" point of view you might be correct. But even if so, I think I'd prefer developer experience over a few byte of memory usage in the client.

And by developer experience I mean: have the function as close as possible to where it's being used.

Unless it's being shared by multiple components: in that case it'd make sense to "pull it out" into a parent component of some sort.

But again: In this case this function is only being used in one specific place. So put it as close as possible to that place.